os/arch/arm/src/amebad/amebad_reboot_reason.c

```diff
@@ -56,26 +56,50 @@
 #include <tinyara/config.h>
 
 #include <tinyara/reboot_reason.h>
+#include "ameba_soc.h"
 /****************************************************************************
  * Public functions
  ****************************************************************************/
 #ifdef CONFIG_SYSTEM_REBOOT_REASON
+
+static reboot_reason_code_t backup_reg;
+
 void up_reboot_reason_init(void)
 {
-	/* If chip vendor needs to initialize for using reboot reason
-	 * chip specific initialization code will be here.
-	 */
+	/* Read the same backup register for the boot reason */
+	backup_reg = BKUP_Read(BKUP_REG1);
```

Comment:
1. After reading, it should be clear.
2. This init function should be called from some initialization function like board_initialize.